crates/topos-sequencer-subnet-runtime/src/proxy.rs

```diff
@@ -142,51 +151,88 @@ impl SubnetRuntimeProxy {
+                                        break None;
+                                    }
+                                }
+                            }
+                            _ = block_task_shutdown.recv() => {
+                                break None;
+                            }
+                        }
+                    };
+                    let current_subnet_block_number = current_subnet_block_number
+                        .expect("need valid subnet block number to start syncing");
 
-                                    // Update certificate block history
-                                    certification.append_blocks(vec![block_info]);
+                    if latest_acquired_subnet_block_number == current_subnet_block_number {
+                        info!(
+                            "Finished synchronization of blocks, latest block received is {}",
+                            latest_acquired_subnet_block_number
+                        );
+                        break;
+                    }
 
-                                    let new_certificates = match certification.generate_certificates().await {
-                                        Ok(certificates) => certificates,
-                                        Err(e) => {
-                                            error!("Unable to generate certificates: {e}");
-                                            break None;
-                                        }
-                                    };
+                    info!(
+                        "Latest retrieved subnet block is {}, current subnet block is {}",
+                        latest_acquired_subnet_block_number, current_subnet_block_number
+                    );
+                    // Sync historical blocks
+                    while latest_acquired_subnet_block_number < current_subnet_block_number {
+                        let next_block_number = latest_acquired_subnet_block_number + 1;
+                        info!("Retrieving historical block {}", next_block_number);
+                        if let Err(e) = SubnetRuntimeProxy::process_next_block(
+                            runtime_proxy.clone(),
+                            &mut subnet_listener,
+                            certification.clone(),
+                            next_block_number,
+                        )
+                        .await
+                        {
+                            panic!("Unable to perform subnet block sync: {}, closing", e);
+                        } else {
+                            latest_acquired_subnet_block_number = next_block_number;
+                        }
 
-                                    debug!("Generated new certificates {new_certificates:?}");
+                        // Give it a little rest for other threads to do their job
+                        tokio::time::sleep(Duration::from_millis(20)).await;
```

Comment:
If we need to sync 100000 blocks, we do not want to run loop as fast as it can, we need to give time to other parts of the process to finish their job